eoscompanion/routes.py

```diff
@@ -950,36 +953,35 @@ def _on_got_metadata_callback(load_metadata_error, metadata_bytes):
             clientId=query['deviceUUID']
         )
     )
-
-    load_record_from_engine_async(Eknc.Engine.get_default(),
-                                  query['applicationId'],
-                                  query['contentId'],
-                                  'metadata',
-                                  _on_got_metadata_callback)
     server.pause_message(msg)
+    content_db_conn.shards_for_application(app_id=query['applicationId'],
+                                           callback=_on_got_shards_callback)
 
 
-def search_single_application(app_id=None,
+def search_single_application(content_db_conn,
+                              app_id=None,
                               tags=None,
                               limit=None,
                               offset=None,
                               search_term=None,
                               callback=None):
-    '''Use EkncEngine.Query to run a query on a single application.
+    '''Use :content_db_conn: to run a query on a single application.
 
     If :tags: is not passed, then we search over the default subset of
     all articles and media objects. Otherwise we search over those
     sets.
     '''
-    query = Eknc.QueryObject(app_id=app_id,
-                             tags_match_any=tags or [
-			                              'EknArticleObject',
-			                              'EknSetObject'
-			                          ],
-			                          limit=limit or _SENSIBLE_QUERY_LIMIT,
-			                          offset=offset or 0,
-			                          search_terms=search_term)
-    Eknc.Engine.get_default().query(query, None, callback)
+    content_db_conn.query(app_id,
+                          query={
+                              'tags-match-any': tags or [
+                                  'EknArticleObject',
+                                  'EknSetObject'
+                              ],
+                              'limit': limit or _SENSIBLE_QUERY_LIMIT,
+                              'offset': offset,
```

Comment:
Should this not be `offset or 0` like in the previous code, or does `query()` take care of that?

Reply:
Good point. I am not sure if passing `None` to the d-bus interface will get silently ignored, so better to use `offset or 0`